File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> targetdiff;
        vector<int> result;
        for(int i = 0; i < nums.size(); i++) {
            if(targetdiff.find(nums[i])!= targetdiff.end()){
                result.push_back(targetdiff[nums[i]]);
                result.push_back(i);
                return result;
            }
            int diff = target - nums[i];
            targetdiff[diff] = i;
        }
        return result;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        start_index = 0
        end_index = len(nums) - 1
        d = dict()
        for index, num in enumerate(nums):
            if num in d :
                result.append(index)
                result.append(d[num])
                return result
            d[target - num] = index

